### PR TITLE
Bridge Baggage

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/973

This turned out to be much smaller than I expected

### CODE REVIEW

File: instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java [hunk 1]
```diff
@@ -156,9 +199,15 @@ public AgentContextWrapper(
         return new AgentContextWrapper(
             agentContext.with(AGENT_SPAN_CONTEXT_KEY, agentSpan), applicationContext);
       }
+      if (k1 == APPLICATION_BAGGAGE_CONTEXT_KEY) {
+        Baggage applicationBaggage = (Baggage) v1;
+        io.opentelemetry.api.baggage.Baggage agentBaggage =
+            BaggageBridging.toAgent(applicationBaggage);
+        System.out.println(((Baggage) v1).asMap());
```
Comment (trask) [inline instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java:206]:
```suggestion
```

File: instrumentation/opentelemetry-api-1.0/javaagent/src/test/groovy/ContextBridgeTest.groovy [hunk 2]
```diff
@@ -80,6 +82,56 @@ class ContextBridgeTest extends AgentTestRunner {
...
+        get("country").entryMetadata == io.opentelemetry.api.baggage.BaggageEntryMetadata.create("asia")
+      }
+
+      agentBaggage = io.opentelemetry.api.baggage.Baggage.builder()
+        .put("country", "italy", io.opentelemetry.api.baggage.BaggageEntryMetadata.create("europe"))
+        .build()
+      agentBaggage.makeCurrent().withCloseable {
+        def updatedApplicationBaggage = Baggage.current()
+        updatedApplicationBaggage.asMap().with {
+          println it
```
Comment (trask) [inline instrumentation/opentelemetry-api-1.0/javaagent/src/test/groovy/ContextBridgeTest.groovy:109]:
```suggestion
```

File: instrumentation/opentelemetry-api-1.0/javaagent/src/test/groovy/ContextBridgeTest.groovy [hunk 3]
```diff
@@ -80,6 +82,56 @@ class ContextBridgeTest extends AgentTestRunner {
...
+          get("country").entryMetadata == BaggageEntryMetadata.create("europe")
+        }
+
+        applicationBaggage = applicationBaggage.toBuilder()
+          .put("food", "cabbage")
+          .build()
+        applicationBaggage.makeCurrent().withCloseable {
+          agentBaggage = io.opentelemetry.api.baggage.Baggage.current()
+          agentBaggage.asMap().with {
+            println it
```
Comment (trask) [inline instrumentation/opentelemetry-api-1.0/javaagent/src/test/groovy/ContextBridgeTest.groovy:123]:
```suggestion
```

File: instrumentation/opentelemetry-api-1.0/javaagent/src/test/groovy/ContextBridgeTest.groovy [hunk 4]
```diff
@@ -80,6 +82,56 @@ class ContextBridgeTest extends AgentTestRunner {
...
+    expect:
+    def applicationBaggage = Baggage.builder()
+      .put("food", "cheese")
+      .put("country", "japan", BaggageEntryMetadata.create("asia"))
+      .build()
+
+    applicationBaggage.makeCurrent().withCloseable {
+      def agentBaggage = io.opentelemetry.api.baggage.Baggage.current()
+      agentBaggage.asMap().with {
+        println it
```
Comment (trask) [inline instrumentation/opentelemetry-api-1.0/javaagent/src/test/groovy/ContextBridgeTest.groovy:95]:
```suggestion
```

**Review by trask — APPROVED** [review]
👍

**Review by anuraaga — COMMENTED** [review]
Oops - forgot to add my check mark, but LGTM

File: instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java]: Oh, I forgot about these 🤦‍♂️

**Pull request merged**